Gem/Code/Source/Components/NetworkWeaponsComponent.cpp

```diff
@@ -68,31 +116,68 @@ namespace MultiplayerSample
 
     void NetworkWeaponsComponent::OnWeaponActivate([[maybe_unused]] const WeaponActivationInfo& activationInfo)
     {
+        // If we're replaying inputs then early out
+        if (GetNetBindComponent()->IsReprocessingInput())
+        {
+            return;
+        }
+
         if (cl_WeaponsDrawDebug && m_debugDraw)
         {
             m_debugDraw->DrawSphereAtLocation
             (
                 activationInfo.m_activateEvent.m_initialTransform.GetTranslation(),
-                0.5f, 
+                0.25f, 
                 AZ::Colors::GreenYellow,
                 10.0f
             );
 
             m_debugDraw->DrawSphereAtLocation
             (
                 activationInfo.m_activateEvent.m_targetPosition,
-                0.5f,
+                0.25f,
                 AZ::Colors::Crimson,
                 10.0f
             );
         }
     }
 
-    void NetworkWeaponsComponent::OnWeaponPredictHit([[maybe_unused]] const WeaponHitInfo& hitInfo)
+    void NetworkWeaponsComponent::OnWeaponPredictHit(const WeaponHitInfo& hitInfo)
     {
+        // If we're replaying inputs then early out
+        if (GetNetBindComponent()->IsReprocessingInput())
+        {
+            return;
+        }
+
+        for (uint32_t i = 0; i < hitInfo.m_hitEvent.m_hitEntities.size(); ++i)
+        {
+            const HitEntity& hitEntity = hitInfo.m_hitEvent.m_hitEntities[i];
+
+            if (cl_WeaponsDrawDebug && m_debugDraw)
+            {
+                m_debugDraw->DrawSphereAtLocation
+                (
+                    hitEntity.m_hitPosition,
+                    1.0f,
+                    AZ::Colors::OrangeRed,
+                    10.0f
+                );
+            }
+
+            AZLOG
+            (
+                NET_Weapons,
+                "Predicted hit on entity %u at position %f x %f x %f",
+                hitEntity.m_hitNetEntityId,
+                hitEntity.m_hitPosition.GetX(),
```

Comment:
Fyi, you can also use AZ::Vector3().ToString().c_str() here with %s.

Reply:
I'll have to remember that, useful.. thanks!